TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -325,6 +325,34 @@ public virtual DataType PrimitiveDataType
             }
         }
 
+        private List<DataType> _compatiblePrimitiveDataTypes;
+        public virtual List<DataType> CompatiblePrimitiveDataTypes
```

Comment:
After second thought, the problem is more general and should be: "which data type is compatible with another one".
I doesn't only apply to PrimitiveDataType.
Eg. Can you move a DataType.AlphanumericEdited to a DataType.Alphanumeric ?

For this issue #1541, we could limit the problem to retrieve the PrimitiveDataType and only use this one for completion. So this could be done for v1.4

Then we can discuss on how to handle compatible DataType (in v1.5)
Maybe we should wait for new SemanticDomain to do that.

So I suggest to :
- Fill property PrimitiveDataType of types Date and Currency when we instantiate them. This will avoid `if (DataType == DataType.Date)`
- Only use property PrimitiveDataType and DataType.Alphanumeric in completion.

Reply:
See latest changes.